builder/vsphere/supervisor/step_watch_source.go

```diff
@@ -175,21 +181,33 @@ func (s *StepWatchSource) getVMIngressIP(ctx context.Context, logger *PackerLogg
 	logger.Info("Getting source VM ingress IP from the VMService object")
 
 	vmServiceObj := &vmopv1alpha1.VirtualMachineService{}
-	err := s.KubeWatchClient.Get(ctx, client.ObjectKey{
+	vmServiceObjKey := client.ObjectKey{
 		Namespace: s.Namespace,
 		Name:      s.SourceName,
-	}, vmServiceObj)
-
-	if err != nil {
-		logger.Error("Failed to get the VMService object in Supervisor cluster")
-		return "", err
 	}
 
-	ingress := vmServiceObj.Status.LoadBalancer.Ingress
-	if len(ingress) == 0 || ingress[0].IP == "" {
-		return "", fmt.Errorf("VMService object's ingress IP is empty")
-	}
+	ticker := time.NewTicker(5 * time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			if err := s.KubeWatchClient.Get(ctx, vmServiceObjKey, vmServiceObj); err != nil {
+				logger.Error("Failed to get the VMService object in Supervisor cluster")
+				continue
+			}
 
-	logger.Info("Successfully retrieved the source VM ingress IP: %s", ingress[0].IP)
-	return ingress[0].IP, nil
+			ingress := vmServiceObj.Status.LoadBalancer.Ingress
+			if len(ingress) == 0 || ingress[0].IP == "" {
+				logger.Info("VMService object's ingress IP is empty, continue checking...")
+				continue
+			}
+
+			logger.Info("Successfully retrieved the source VM ingress IP: %s", ingress[0].IP)
+			return ingress[0].IP, nil
+
+		case <-s.timeoutCh:
```

Comment:
To ensure each step gets the full elapsed time. 
```suggestion
		case <-time.After(time.Duration(s.Config.WatchSourceTimeoutSec) * time.Second):
```

Reply:
Ditto.